python/plugins/processing/algs/gdal/Buffer.py

```diff
@@ -134,8 +134,15 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         source_details = self.getOgrCompatibleSource(
             self.INPUT, parameters, context, feedback, executing
         )
+        if not source_details.layer_name:
```

Comment:
Are these changes (and all the others below, except the virtual layers part) related to the PR title "fix running gdal algs with virtual layers"  ?

Reply:
Yes